pkg/securitypolicy/securitypolicyenforcer_rego.go

```diff
@@ -21,50 +36,491 @@ func init() {
 	defaultEnforcer = regoEnforcer
 }
 
+//go:embed framework.rego
+var FrameworkCode string
+
+//go:embed policy.rego
+var PolicyCode string
+
+var Indent string = "    "
+
 // RegoEnforcer is a stub implementation of a security policy, which will be
 // based on [Rego] policy language. The detailed implementation will be
 // introduced in the subsequent PRs and documentation updated accordingly.
 //
 // [Rego]: https://www.openpolicyagent.org/docs/latest/policy-language/
-type RegoEnforcer struct{}
+type RegoEnforcer struct {
+	// Rego which describes policy behavior (see above)
+	behavior string
+	// Rego which describes policy objects (containers, etc.)
+	objects string
+	// Mutex to prevent concurrent access to fields
+	mutex *sync.Mutex
+	// Rego data object, used to store policy state
+	data map[string]interface{}
+	// Base64 encoded (JSON) policy
+	base64policy string
+	// Compiled modules
+	compiledModules *ast.Compiler
+}
 
-var (
-	_                 SecurityPolicyEnforcer = (*RegoEnforcer)(nil)
-	ErrNotImplemented                        = errors.New("not implemented")
-)
+var _ SecurityPolicyEnforcer = (*RegoEnforcer)(nil)
 
-func createRegoEnforcer(_ SecurityPolicyState, _, _ []oci.Mount) (SecurityPolicyEnforcer, error) {
-	return &RegoEnforcer{}, nil
+type securityPolicyInternal struct {
+	AllowAll   bool
+	Containers []*securityPolicyContainer
 }
 
-func (RegoEnforcer) EnforceDeviceMountPolicy(_, _ string) error {
-	return ErrNotImplemented
+func (sp SecurityPolicy) toInternal() (*securityPolicyInternal, error) {
+	policy := new(securityPolicyInternal)
+	var err error
+	if policy.Containers, err = sp.Containers.toInternal(); err != nil {
+		return nil, err
+	}
+
+	return policy, nil
```

Comment:
We seem to be missing copying over `sp.AllowAll`.

Reply:
Yes it isnt currently used as part of the rego policy. As part of the json -> rego, the AllowAll is only used for creating an OpenDoorEnforcer.

Reply:
Can we remove it on `securityPolicyInternal` then? I think we should either remove it, or correctly set it based on the input.

Reply:
@SeanTAllen  is not technically correct here. `AllowAll` is currently used to set an `allow_all` rule in the generated policy which in turn allows the Rego policy to support an `AllowAll` policy, in preparation for the other code path going away. If we want to handle that another way, we can, but it is used at the moment.

Reply:
(I've added in the missing AllowAll copy for now)